### ISSUE TITLE
Cancel the Task of OpenAsync will throw MySqlConnector.MySqlException("Connect Timeout expired.")

### ISSUE THREAD

**Issue by Flithor** [mysql-net/MySqlConnector#931]
```
CancellationTokenSource cancellationTokenSource;
public async Task StartTryConnection()
{
    //var dbConnection = new SqlConnection("server=128.0.0.1;userid=sa;");
    var dbConnection = new MySqlConnection("server=128.0.0.1;userid=root;");
    cancellationTokenSource = new CancellationTokenSource();
    await dbConnection.OpenAsync(cancellationTokenSource.Token)
        .ContinueWith(task =>
        {
            dbConnection.Close();
            if (task.IsCanceled)
            {
                //Cancel the task of SqlConnection.OpenAsync will go here
                MessageBox.Show("Canceled!");
            }
            else if (task.IsCompletedSuccessfully)
            {
                MessageBox.Show("Success!");
            }
            else
            {
                //throw MySqlConnector.MySqlException("Connect Timeout expired.") in task and will go here
                MessageBox.Show(task.Exception.InnerException.Message);
            }
        });
}

public void StopTryConnection()
{
    cancellationTokenSource?.Cancel();
}
```

I think task status should be the `TaskStatus.Canceled`
Related code here:
https://github.com/mysql-net/MySqlConnector/blob/cf481225967a1cf2a2a3303b5193d6c9a7730e24/src/MySqlConnector/MySqlConnection.cs#L862

**Comment by bgrainger** [comment]
I agree that cancelling the `CancellationToken` passed into the method should throw a `TaskCanceledException`. (I believe it used to do this, but don't know if/when that changed.)

**Comment by bgrainger** [comment]
Fixed in [1.3.0](https://github.com/mysql-net/MySqlConnector/releases/tag/1.3.0).
